x-pack/plugins/lens/public/visualizations/xy/xy_config_panel/layer_header.tsx

```diff
@@ -62,13 +78,33 @@ function ReferenceLayerHeader() {
   );
 }
 
-function AnnotationsLayerHeader() {
+function AnnotationsLayerHeader({
+  title,
+  hasUnsavedChanges,
+}: {
+  title: string | undefined;
+  hasUnsavedChanges: boolean;
+}) {
   return (
     <StaticHeader
       icon={IconChartBarAnnotations}
-      label={i18n.translate('xpack.lens.xyChart.layerAnnotationsLabel', {
-        defaultMessage: 'Annotations',
-      })}
+      label={
+        title ||
+        i18n.translate('xpack.lens.xyChart.layerAnnotationsLabel', {
+          defaultMessage: 'Annotations',
+        })
+      }
+      indicator={
+        hasUnsavedChanges && (
+          <EuiIconTip
```

Comment:
Would it be possible to add 4px of space between the annotation layer's title and this dirty state indicator?

Reply:
Addressed in https://github.com/elastic/kibana/pull/159057